src/avram/save_operation.cr

```diff
@@ -59,6 +51,18 @@ abstract class Avram::SaveOperation(T)
     @params = Avram::Params.new
   end
 
+  def database
+    T.database
+  end
+
+  def table_name
+    T.table_name
+  end
+
+  def primary_key_name
+    T.primary_key_name
+  end
```

Comment:
I think this might help clarify what it is doing and take a bit less space 

```suggestion
  delegate :database, :table_name, :primary_key_name, to: T
```